Source/WebKit/UIProcess/WebPageProxy.cpp

```diff
@@ -5258,7 +5258,9 @@ void WebPageProxy::findStringMatches(const String& string, OptionSet<FindOptions
         return;
     }
 
-    send(Messages::WebPage::FindStringMatches(string, options, maxMatchCount));
+    sendWithAsyncReply(Messages::WebPage::FindStringMatches(string, options, maxMatchCount), [this, protectedThis = Ref { *this }](String string, Vector<Vector<WebCore::IntRect>> matches, int32_t firstIndexAfterSelection) {
```

Comment:
Just capture string in the lambda instead of expecting the web content process to send back the exact same string.